<launch>
    <arg name="marker_size" default="2.4" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.02" />

    <arg name="camera_frame_id" default="picam_global" />
    <arg name="cam_image_topic" default="/$(arg camera_frame_id)/image_raw" />
    <arg name="cam_info_topic" default="/$(arg camera_frame_id)/camera_info" />
    <arg name="output_frame" default="$(arg camera_frame_id)" />

    <arg name="video_device" default="/dev/video3" />
    <arg name="calib_url" default="file://${HOME}/.ros/camera_info/picam_global.yaml" />
    <arg name="framerate" default="5" />

    <param
        name="camera_frame"
        value="$(arg camera_frame_id)"/>
    <param
        name="camera_index"
        value="$(arg video_device)" />

    <node
        pkg="tf"
        type="static_transform_publisher"
        args="0 0 1 0 -90 0 world $(arg camera_frame_id) 100"
        name="world2pi" />

    <node
        pkg="rpi_tracker"
        type="rpi_usb_cam.py"
        name="picam_global">
        <param
            name="camera_frame_id"
            value="$(arg camera_frame_id)" />
        <param
            name="video_device"
            value="$(arg video_device)"/>
        <param
            name="image_height"
            value="720" />
        <param
            name="image_width"
            value="1280" />
        <param
            name="framerate"
            value="$(arg framerate)" />

    </node>

    <node
        pkg="ar_track_alvar"
        type="individualMarkersNoKinect"
        name="ar_track_alvar"
        respawn="true"
        output="screen">

        <param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="/world" />

        <remap from="camera_image" to="$(arg cam_image_topic)"/>
        <remap from="camera_info" to="$(arg cam_info_topic)"/>
    </node>

</launch>